enterprise/server/test/integration/remote_bazel/remote_bazel_test.go

```diff
@@ -0,0 +1,208 @@
+	elpb "github.com/buildbuddy-io/buildbuddy/proto/eventlog"
+	inpb "github.com/buildbuddy-io/buildbuddy/proto/invocation"
+	uidpb "github.com/buildbuddy-io/buildbuddy/proto/user_id"
+)
+
+func TestWithPublicRepo(t *testing.T) {
+	// Use a dir that is persisted on recycled runners
+	rootDir := "/root/workspace/remote-bazel-integration-test"
+	err := os.Setenv("HOME", rootDir)
+	require.NoError(t, err)
+
+	err = os.MkdirAll(rootDir, 0755)
+	require.NoError(t, err)
+
+	if _, err := os.Stat(fmt.Sprintf("%s/bazel-gazelle", rootDir)); os.IsNotExist(err) {
+		output := testshell.Run(t, rootDir, "git clone https://github.com/bazelbuild/bazel-gazelle --filter=blob:none --depth=1")
+		require.NotContains(t, output, "fatal")
+	}
+
+	err = os.Chdir(fmt.Sprintf("%s/bazel-gazelle", rootDir))
+	require.NoError(t, err)
+
+	// Run a server and executor locally to run remote bazel against
+	env, bbServer, _ := runLocalServerAndExecutor(t, "")
+	ctx := env.WithUserID(context.Background(), env.UserID1)
+	reqCtx := &ctxpb.RequestContext{
+		UserId:  &uidpb.UserId{Id: env.UserID1},
+		GroupId: env.GroupID1,
+	}
+
+	// Get an API key to authenticate the remote bazel request
+	bbClient := env.GetBuildBuddyServiceClient()
+	apiRsp, err := bbClient.CreateApiKey(ctx, &akpb.CreateApiKeyRequest{
+		RequestContext: reqCtx,
+		GroupId:        env.GroupID1,
+		Capability: []akpb.ApiKey_Capability{
+			akpb.ApiKey_CAS_WRITE_CAPABILITY,
+			akpb.ApiKey_CACHE_WRITE_CAPABILITY,
+			akpb.ApiKey_ORG_ADMIN_CAPABILITY,
+		},
+	})
+	require.NoError(t, err)
+	apiKey := apiRsp.ApiKey.Value
+
+	exitCode, err := remotebazel.HandleRemoteBazel([]string{
+		fmt.Sprintf("--remote_runner=%s", bbServer.GRPCAddress()),
+		// Have the ci runner use the "none" isolation type because it's simpler
+		// to setup than a firecracker runner
+		"--runner_exec_properties=workload-isolation-type=none",
+		"--runner_exec_properties=container-image=",
+		// Give more memory to the CI runner because it needs to clone the gazelle repo
+		"--runner_exec_properties=EstimatedComputeUnits=3",
+		"help",
+		fmt.Sprintf("--remote_header=x-buildbuddy-api-key=%s", apiKey)})
+	require.NoError(t, err)
+	require.Equal(t, 0, exitCode)
+
+	// Check the invocation logs to ensure the "bazel help" command successfully run
+	searchRsp, err := bbClient.SearchInvocation(ctx, &inpb.SearchInvocationRequest{
+		RequestContext: reqCtx,
+		Query:          &inpb.InvocationQuery{GroupId: env.GroupID1},
+	})
+	require.NoError(t, err)
+	require.Equal(t, 1, len(searchRsp.GetInvocation()))
+	invocationID := searchRsp.Invocation[0].InvocationId
+
+	logResp, err := bbClient.GetEventLogChunk(ctx, &elpb.GetEventLogChunkRequest{
+		InvocationId: invocationID,
+		MinLines:     math.MaxInt32,
+	})
+	require.NoError(t, err)
+	require.Contains(t, string(logResp.GetBuffer()), "Usage: bazel <command> <options>")
+}
+
+func TestWithPrivateRepo(t *testing.T) {
+	repoName := "private-test-repo"
+	username := "maggie-lou"
```

Comment:
Decided that it will probably be more annoying to track a password for a service account. Anyone can re-generate this PAT if they need, even if I leave the company and we can't access this token or something